src/ConnectedMode/ServerQueryInfoProvider.cs

```diff
@@ -0,0 +1,70 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+
+namespace SonarLint.VisualStudio.ConnectedMode
+{
+    /// <summary>
+    /// Returns the information necessary to make various queries to the Sonar server
+    /// </summary>
+    /// <remarks>Facade to simplify classes that need to make branch-aware calls.
+    /// The projectKey and server branch are provided by different interfaces. This interface
+    /// simplifies getting the information.</remarks>
+    internal interface IServerQueryInfoProvider
+    {
+        /// <summary>
+        /// Returns the projectKey and branch name for the current bound solution, or
+        /// (null, null) if the solution is not bound
+        /// </summary>
+        Task<(string projectKey, string branchName)> GetProjectKeyAndBranchAsync(CancellationToken token);
```

Comment:
The web calls to fetch the suppressions need to know the current project key and branch name, and need to handle the details not being available.

Originally, I'd put this functionality in the `SuppressionIssueStoreUpdater` class but it made it more complicated than necessary, so I broke it out into a separate class. It's likely that we could re-use it to simpligy other classes that call the server.